### PR TITLE
pynfs: Add soft failure

### PR DESCRIPTION
**Pull request by pevik**
bsc#1192211 (needed as tests are using `autotest::loadtest()`).

verification run:
- http://quasar.suse.cz/tests/7972#step/LOCK24/2 http://quasar.suse.cz/tests/7971#step/LOCK24/2 (soft failure)
- http://quasar.suse.cz/tests/7973#step/generate_report/18 (all passed)

### CODE REVIEW

**Comment by pevik** [comment]
NOTE for all passed, there could be `tests/nfs/pynfs_failed.pm` instance writing "ALL PASS" to make it clear.

**Comment by pevik** [comment]
@czerw @SeroSun @lansuse can you please review?

File: tests/nfs/generate_report.pm [hunk 1]
```diff
@@ -50,6 +51,11 @@ sub display_results {
         my $targs = OpenQA::Test::RunArgs->new();
         $targs->{data} = $test;
         autotest::loadtest("tests/nfs/pynfs_failed.pm", name => $test->{code}, run_args => $targs);
+        $fail = 1;
+    }
+
+    if (!$fail) {
+        record_info('ALL PASSED');
```
Comment (czerw) [inline tests/nfs/generate_report.pm:58]: record_info('ALL PASSED') if (!$fail);

Comment (pevik) [reply tests/nfs/generate_report.pm]: Sure, fixed :). I just wonder if wouldn't be more obvious for test result using tests/nfs/pynfs_failed.pm instance writing "ALL PASS".

File: tests/nfs/pynfs_failed.pm [hunk 2]
```diff
@@ -14,13 +14,21 @@ use Data::Dumper;
...
     die 'failure does not exist' unless (exists($data->{failure}));
 
-    $self->{result} = 'fail';
-    record_info('ERROR', "$data->{failure}->{message}\n\n$data->{failure}->{err}\n");
+    $msg = "$data->{failure}->{message}\n\n$data->{failure}->{err}";
+
+    if ($data->{code} eq 'LOCK24' && $data->{failure}->{message} eq
+        'OP_LOCK should return NFS4_OK, instead got NFS4ERR_BAD_SEQID') {
+        $self->record_soft_failure_result("LOCK24 failure is known, verriding to softfail: bsc#1192211\n\n" . $msg);
+    } else {
```
Comment (czerw) [inline tests/nfs/pynfs_failed.pm:28]: I this approach in tuned  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/f1f28df67f1588f50f5a8c2529a908343e3e2f9b/tests/kernel/tuned.pm#L29 . Couldn't it be better ? There is only one fail for SP4 but for older there is at least one more.

**Review by czerw — APPROVED** [review]
LGTM for SP4

**Comment by lansuse** [comment]
LGTM

**Comment by SeroSun** [comment]
By network issue, I can't open the file changes. But by looking into those verify run, I assume this is a nice enhancement :)

**Pull request merged**